### ISSUE TITLE
Please document these files

### ISSUE THREAD

**Issue by pgrosu** [google/deepvariant#87]
Congratulations on the new release, and glad to see that Intel MKL was utilized for optimized processing.

When you have a chance, could you please document the following and provide use-cases for new users:

https://github.com/google/deepvariant/blob/r0.7/cloudbuild.yaml

https://github.com/google/deepvariant/blob/r0.7/cloudbuild_CBI.yaml

Thanks,
~p

**Comment by pichuan** [comment]
Hi Paul,
there is actually some explanation in the file (below the Copyright lines):
https://github.com/google/deepvariant/blob/r0.7/cloudbuild.yaml#L28-L34

We use these yaml files to build and push docker images. You probably could re-use them to build and push DeepVariant docker images to projects where you have access to.

We didn't document these more, because we didn't think it's a very common use case.
If it's confusing to have these files on the top level, I can check with my colleagues and see if we can move them to a sub-directory in the next release. Would that help?

**Comment by pgrosu** [comment]
Hi Pichuan,

I understand the nature of Cloud Build, and has a very practical use-case for many users that would like to automate their analysis.  I'm sort of hinting at something else.  So just imagine you are pitching this to some Bioinformatician that has been doing their analysis on an internal cluster for years.  S/he has heard from others of the benefits of DeepVariant through Google Cloud resources, and now has a little bit of free time to try it out to convince their boss to use it.  The idea is that the root directory should be as simple as possible, so users don't get overwhelmed or confused - with all the other directories in the tree flowing naturally from it.  A minimal number of obvious scripts (with associated documentation) should naturally transition them from source code, through the build phase, and as quickly to the analysis process.  Basically less is more, and you do not want to not add more to the main directory as you begin to approach version 1.0.

So ideally what would be nice is to have one build script - let's call it `build` - that reads a `config/build` directory - and does the following:

1. Analyzes hidden build-generated files under `.build/...`, and detects if the `prereq` and `build-and-test` phases have been successfully processed.  Otherwise, it would report back to the user the build status, and query the user with a recommendation on the next required step to run.  For example:

```
$ build status

Checking DeepVariant build prerequisites... OK
Checking DeepVariant test environment compatibility [GPU]... OK
Checking DeepVariant build binaries... MISSING

Proceed with building binaries [Y/n]?

```

2. Once the builds have been successfully completed, then it would tell the user the available options to run (i.e. `make_examples`, `call_variants`, `postprocess_variants`, `pkrusche/hap.py`, etc.) as well as any analysis that has been completed if a `results` or `analysis-results` folder is present.

3. For Cloud Build, that would also be part of the `build` script, when they provide the `help` argument:

```
$ ./build help
...
cloudbuild   Builds Docker images of DeepVariant, and 
             pushes them on the Google Container Registry (gcr.io) 
...

$ ./build cloudbuild help
CPU     Builds a DeepVariant Docker image for CPU usage.
GPU     Builds a DeepVariant Docker image for GPU usage.
Runner  Builds a DeepVariant Docker image for large-scale analysis run
          using the Genomics Pipelines API.

$
```

Even `Runner` is a bit too general, so maybe calling it `LargeScaleAnalysis`, or something which should be instantly recognizable as to its intended use.

Hope it helps,
Paul

**Comment by nmousavi** [comment]
For the use case you mentioned, one should just use published docker images (see images pointed in https://cloud.google.com/genomics/docs/tutorials/deepvariant).

These config files are meant to be used by cloud build. See "Build using a build config file" in https://cloud.google.com/cloud-build/docs/quickstart-docker for how they are used. The tool you suggested will be useful once DeepVariant accepts external contributions.

**Comment by pgrosu** [comment]
Hi Nima,

That is a good point, and glad to see the tutorials page updated today.  Though it would have been nice to have it highlight the `--use_tpu` flag option and `0.7.0` version, rather than the `0.6.x`.  The tags in the docker images for the runner are still as a release candidate `0.7.0rc1`, with `0.6.1` as the `latest`.

In any case, you guys are doing great work, and there is quite a lot of expertise on this side of the world as well to expand on DeepVariant's capabilities, and foster an even greater community.  You might want to update the [contributing document](https://github.com/google/deepvariant/blob/master/CONTRIBUTING.md), since it mentions 1H 2018 for external contributions and CI testing on GitHub - and it's almost September.  Version 0.7 seemed to have just dropped out of the sky without any formal review process, and we could have helped you iron out some of the kinks (i.e. such as minor spelling on the [TPU case study](https://github.com/google/deepvariant/blob/r0.7/docs/deepvariant-tpu-training-case-study.md#shuffle-each-set-of-examples-and-generate-a-data-configuration-file-for-each) [s/Optoinal/Optional]).  Your [Docker documentation](https://github.com/google/deepvariant/blob/master/docs/deepvariant-docker.md#run-the-pipeline) still lists your image to use as `0.4.1`, which has an older channel representation of the input data - though using `latest` is a comment that might make a new user not realize the amount of difference between the versions.

Some things seemed a bit rushed, and doing it together would have provided more eyes with expanded tutorials for attracting new users to lower any barriers to entry for new users - who can be quite picky when trying out new software.

Hope it helps,
Paul

**Comment by pichuan** [comment]
Hi @pgrosu , thanks for your feedback!
Thanks to @nmousavi 's work, the Cloud runner page is now updated:
https://cloud.google.com/genomics/docs/tutorials/deepvariant with 0.7.0, and 0.7.0 deepvariant_runner image is now tagged as latest.

In terms of our GitHub page --
I fixed a few small thing such as the typo (and ran spell checking and fixed a few more!) Also fixed the `0.4.1` issue.

I'll also address the contributing document at some point soon.

These changes are right now still just in our internal codebase. I'll get it out when I have a chance to push out some documentation fixes. Thank you!

**Comment by pichuan** [comment]
Hi @pgrosu , I filed an internal issue to track your suggestions in https://github.com/google/deepvariant/issues/87#issuecomment-413745335
I'll need to digest it a bit more, and probably talk to to a few users in more detail to design this experience better.

On a high level, I think I'll need to at least think about overhauling the GitHub page to make it super clear how to run  - even just with our docker image, it's apparently still not obvious for users to find (because it's hidden in the docs).

And, the suggestion of making things as simple as possible is great one - both in terms of the directory structure, and also the output that our programs output. I think I have stared DeepVariant for too long that I'm losing empathy for people who look at it for the first time.

I'll close this issue on GitHub, but will plan to think about this in more detail and hope to get back with at least some more plan later.
